src/main/java/apoc/cypher/Cypher.java

```diff
@@ -434,4 +437,15 @@ private static Collection asCollection(Object value) {
     public Stream<MapResult> doWhenCase(@Name("conditionals") List<Object> conditionals, @Name(value="elseQuery", defaultValue = "") String elseQuery, @Name(value="params", defaultValue = "") Map<String, Object> params) {
         return whenCase(conditionals, elseQuery, params);
     }
+
+    @Procedure(name = "apoc.cypher.runFirstColumnWrites", mode = Mode.WRITE)
+    @Description("apoc.cypher.runFirstColumnWrites(statement, params) - executes write statement with given parameters, yields first column only (nodes)")
+    public Stream<NodeResult> runFirstColumnWrites(@Name("cypher") String statement, @Name("params") Map<String, Object> params) {
+        if (params == null) params = Collections.emptyMap();
+        try (Result result = db.execute(withParamMapping(statement, params.keySet()), params)) {
+            String firstColumn = result.columns().get(0);
+            ResourceIterator<Node> resultColumn = result.columnAs(firstColumn);
+            return resultColumn.stream().map(NodeResult::new);
```

Comment:
what happens if the query doesn't return anything or no nodes?
How should it fail/finish then?
With no results, with a single null result?
Would it make more sense to just have an ObjectResult and have the consumer decide?
Alternatively we could add an optional param to apoc.cypher.run ? that says which column names or indexes to return? We could also use map-deconstruction on 

```
CALL apoc.cypher.run(query,params) yield row
WITH apoc.map.values(row,[index/name])[0]
```

I kinda  want to avoid a too specific procedure.

Btw. please note that this can't be inlined like we do with functions within pattern comprehensions.